src/map.rs

```diff
@@ -10,17 +9,13 @@ use crate::ctx::{AddCtx, ReadCtx, RmCtx};
 use crate::traits::{Causal, CmRDT, CvRDT};
 use crate::vclock::{Actor, Dot, VClock};
 
-/// Key Trait alias to reduce redundancy in type decl.
-pub trait Key: Debug + Ord + Clone {}
-impl<T: Debug + Ord + Clone> Key for T {}
-
 /// Val Trait alias to reduce redundancy in type decl.
-pub trait Val<A: Actor>: Debug + Default + Clone + Causal<A> + CmRDT + CvRDT {}
+pub trait Val<A: Actor>: Clone + Causal<A> + CmRDT + CvRDT {}
```

Comment:
@davidrusu: Something I'm curious about is if `CmRDT` _and_ `CvRDT` are necessary here? To me, this sounds restrictive because I'd imagine you could use one or the other but what if I can't provide both instances?

Reply:
Yeah I think you're right, I think this is here because we share the Val trait across both the `impl CmRDT for Map` and `impl CvRDT for Map`. For the `CmRDT` case we just need `Val: CmRDT`, whereas for the `CvRDT` case we just need `Val: CvRDT`.

Haven't validated this but it sounds about right

Reply:
If you want to make this change as part of this PR, that's cool otherwise happy to just accept the PR as is (with the modification mentioned below about the `Default` constraint)

Reply:
I think I'll leave it for now. I started pulling the thread but I think it'll get hairy and would involve making some decisions about type parameters :grin: I'll create a follow-up ticket for it :ok_hand:

Reply:
https://github.com/rust-crdt/rust-crdt/issues/68